### ISSUE TITLE
[Bug]: mux_client_request_session: session request failed: Session open refused by peerControlSocket

### ISSUE THREAD

**Issue by carlosbensant** [coollabsio/coolify#4173]
### Error Message and Logs

mux_client_request_session: session request failed: Session open refused by peerControlSocket /var/www/html/storage/app/ssh/mux/mux_pcws8oc already exists, disabling multiplexing

### Steps to Reproduce

Logs from coolify container:
2024-Nov-08 04:50:27.918668
Preparing container with helper image: ghcr.io/coollabsio/coolify-helper:1.0.2.
2024-Nov-08 04:50:28.040728
docker rm -f pk4sc8oc0080sk8wsco8wo08
2024-Nov-08 04:50:28.040728
mux_client_request_session: session request failed: Session open refused by peer
2024-Nov-08 04:50:28.136109
ControlSocket /var/www/html/storage/app/ssh/mux/mux_pcws8oc already exists, disabling multiplexing
2024-Nov-08 04:50:28.364171
Error response from daemon: No such container: pk4sc8oc0080sk8wsco8wo08
2024-Nov-08 04:50:28.490634
docker run -d --network exo-production --name pk4sc8oc0080sk8wsco8wo08 --rm -v /root/.docker/config.json:/root/.docker/config.json:ro -v /var/run/docker.sock:/var/run/docker.sock ghcr.io/coollabsio/coolify-helper:1.0.2
2024-Nov-08 04:50:28.490634
mux_client_request_session: session request failed: Session open refused by peer
2024-Nov-08 04:50:28.611967
ControlSocket /var/www/html/storage/app/ssh/mux/mux_pcws8oc already exists, disabling multiplexing
2024-Nov-08 04:50:28.941849
d203a203ea16e1a7e3112cef0732c7b7574a231c5df5f736cf35f8e21a513924
2024-Nov-08 04:50:29.438459
docker exec pk4sc8oc0080sk8wsco8wo08 bash -c 'mkdir -p /artifacts/pk4sc8oc0080sk8wsco8wo08'
2024-Nov-08 04:50:29.438459
mux_client_request_session: session request failed: Session open refused by peer
2024-Nov-08 04:50:29.537888
ControlSocket /var/www/html/storage/app/ssh/mux/mux_pcws8oc already exists, disabling multiplexing
2024-Nov-08 04:50:30.132676
docker exec pk4sc8oc0080sk8wsco8wo08 bash -c 'GIT_SSH_COMMAND="ssh -o ConnectTimeout=30 -p 22 -o Port=22 -o LogLevel=ERROR -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" git ls-remote [https://x-access-token:<REDACTED>@github.com/company/projectname.git](https://x-access-token:%3CREDACTED%3E@github.com/company/projectname.git) main'
2024-Nov-08 04:50:30.132676
mux_client_request_session: session request failed: Session open refused by peer
2024-Nov-08 04:50:30.227492
ControlSocket /var/www/html/storage/app/ssh/mux/mux_pcws8oc already exists, disabling multiplexing
2024-Nov-08 04:50:30.633924
85f96bb9c23cc6a60e001e6be1bfe24aa3c02b41	refs/heads/main
2024-Nov-08 04:50:30.810985
----------------------------------------
2024-Nov-08 04:50:30.816711
Importing company/projectname:main (commit sha HEAD) to /artifacts/pk4sc8oc0080sk8wsco8wo08.
2024-Nov-08 04:50:30.950886
docker exec pk4sc8oc0080sk8wsco8wo08 bash -c 'git clone -b "main" [https://x-access-token:<REDACTED>@github.com/company/projectname.git](https://x-access-token:%3CREDACTED%3E@github.com/company/projectname.git) /artifacts/pk4sc8oc0080sk8wsco8wo08 && cd /artifacts/pk4sc8oc0080sk8wsco8wo08 && GIT_SSH_COMMAND="ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" git submodule update --init --recursive && cd /artifacts/pk4sc8oc0080sk8wsco8wo08 && GIT_SSH_COMMAND="ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null" git lfs pull'
2024-Nov-08 04:50:30.950886
mux_client_request_session: session request failed: Session open refused by peer
2024-Nov-08 04:50:31.052166
ControlSocket /var/www/html/storage/app/ssh/mux/mux_pcws8oc already exists, disabling multiplexing
2024-Nov-08 04:50:31.308317
Cloning into '/artifacts/pk4sc8oc0080sk8wsco8wo08'...
2024-Nov-08 04:50:35.289783
docker exec pk4sc8oc0080sk8wsco8wo08 bash -c 'mkdir -p /artifacts/pk4sc8oc0080sk8wsco8wo08'
2024-Nov-08 04:50:35.289783
mux_client_request_session: session request failed: Session open refused by peer
2024-Nov-08 04:50:35.390775
ControlSocket /var/www/html/storage/app/ssh/mux/mux_pcws8oc already exists, disabling multiplexing
2024-Nov-08 04:50:36.150860
mkdir -p /data/coolify/applications/swcsg80
2024-Nov-08 04:50:36.150860
mux_client_request_session: session request failed: Session open refused by peer
2024-Nov-08 04:50:36.255420
ControlSocket /var/www/html/storage/app/ssh/mux/mux_pcws8oc already exists, disabling multiplexing
2024-Nov-08 04:50:36.510735
docker exec pk4sc8oc0080sk8wsco8wo08 bash -c 'cd /artifacts/pk4sc8oc0080sk8wsco8wo08 && git log -1 mux_client_request_session: session request failed: Session open refused by peerControlSocket /var/www/html/storage/app/ssh/mux/mux_pcws8oc already exists, disabling multiplexing85f96bb9c23cc6a60e001e6be1bfe24aa3c02b41 --pretty=%B'
2024-Nov-08 04:50:36.510735
mux_client_request_session: session request failed: Session open refused by peer
2024-Nov-08 04:50:36.612025
ControlSocket /var/www/html/storage/app/ssh/mux/mux_pcws8oc already exists, disabling multiplexing
2024-Nov-08 04:50:36.843824
fatal: invalid object name 'mux_client_request_session'.
2024-Nov-08 04:50:37.430145
docker rm -f pk4sc8oc0080sk8wsco8wo08 >/dev/null 2>&1
2024-Nov-08 04:50:37.430145
mux_client_request_session: session request failed: Session open refused by peer
2024-Nov-08 04:50:37.532159
ControlSocket /var/www/html/storage/app/ssh/mux/mux_pcws8oc already exists, disabling multiplexing

### Coolify Version

v4.0.0-beta.360

### Are you using Coolify Cloud?

No (self-hosted)

### Operating System and Version (self-hosted)

Ubuntu 22.04.4 LTS (GNU/Linux 6.5.0-1020-aws x86_64)

**Comment by carlosbensant** [comment]
Related: https://github.com/coollabsio/coolify/issues/2282

**Comment by carlosbensant** [comment]
![Image](https://github.com/user-attachments/assets/4ba48bf7-7d08-4381-b21a-1e275854c9eb)

**Comment by gerwitz** [comment]
Also having this problem, and find the `/var/www/html/storage/app/ssh/mux/` path suspicious. Could this be akin to an error Ansible had with `$HOME`?

See also #3716

**Comment by oliverfrost1** [comment]
+1. I'm getting this same issues on two unrelated services running on the same server. I have not been able to find a fix. I'm on v4 beta 380.

**Comment by carlosbensant** [comment]
I solved the issues by restarting the servers (and reinitializing docker).

**Comment by oliverfrost1** [comment]
> I solved the issues by restarting the servers (and reinitializing docker).

Thank you for this.

I was able to fix it just by running "sudo systemctl restart docker" without restarting the machine.

**Comment by gerwitz** [comment]
This is still a recurring problem for me, resolvable by removing the contents of `/data/coolify/ssh/mux` (no need to restart anything)

**Comment by vojtech-cerveny** [comment]
For me, it seems that if I had few deployments in short time, this problem appeared. But I removed all from `/data/coolify/ssh/mux` as @gerwitz suggested, and it resolved problem.

**Comment by Cristy94** [comment]
Still happening in  4.0.0-beta.420, I have to delete the mux after every deployment.

**Comment by Cristy94** [comment]
This is a HUGE issue!

Half of my containers/websites stopped last night, and they didn't restart because of this mux issue which stops deployments!

**Comment by itxtoledo** [comment]
same problem here, i cannot deploy anything

**Comment by xorweak** [comment]
same problem here.

**Comment by sschwarz-allbuyone** [comment]
We are experiencing the exact same mux issue after every release. This only started happening after updating Coolify to the latest version (from v4.0.0-beta.416). We also have to manually delete the mux folder each time to get deployments working again.
Has anyone found a different workaround or a more permanent fix for this problem?

**Comment by body20002** [comment]
Hello, I'm experiencing the same issue here on version `v4.0.0-beta.420.6`
How can I get my containers running again?

**Comment by suptejas** [comment]
Experiencing this issue in the latest version too. Is a permanent fix for this planned? Thanks!

**Comment by peaklabs-dev** [comment]
Yes. v5 will no longer use multiplexing which will eliminate this issue https://github.com/coollabsio/coolify/issues/3226#issuecomment-3057687967

**Comment by ikwerre-dev** [comment]
when is v5 coming out?

**Comment by peaklabs-dev** [comment]
@ikwerre-dev In a couple of months as it is an almost complete rewrite.
